/**:
  ros__parameters:
    controller_frequency: 50.0
    initial_perception_timeout: 5.0
    wait_charge_timeout: 5.0
    dock_approach_timeout: 20.0
    undock_linear_tolerance: 0.08
    undock_angular_tolerance: 0.08
    max_retries: 3
    base_frame: "<robot_namespace>/base_link"
    fixed_frame: "<robot_namespace>/odom"
    dock_backwards: false
    dock_prestaging_tolerance: 0.5


    dock_plugins: ["panther_docking_dock"]
    panther_docking_dock:
      plugin: panther_docking::PantherChargingDock
      base_frame: "<robot_namespace>/base_link"
      docking_distance_threshold: 0.15
      docking_yaw_threshold: 0.15
      staging_x_offset: -0.5
      staging_yaw_offset: 0.0

      # TODO: @delihus Try to remove this parameters by using docking station description in the ros_components_description
      # Transform between april tag frame and dock pose. An april tag Z+ faces always a camera
      external_detection_timeout: 0.3
      external_detection_translation_x: 0.0
      external_detection_translation_y: -0.175 # Distance between the detection and ground
      external_detection_translation_z: 0.8 # Distance between the detection and the front of the robot
      external_detection_rotation_roll: -1.57
      external_detection_rotation_pitch: 0.0
      external_detection_rotation_yaw:  1.57
      filter_coef: 0.1

      enable_charger_service_call_timeout: 1.0

    docks: ["main_dock"]
    main_dock:
      type: "panther_docking_dock"
      frame: <robot_namespace>/main_dock
      pose: [0.0, 0.0, 0.0] # position of the dock device (not the staging position), the front (X+) of the dock should point away from the robot

    controller:
      k_phi: 1.0
      k_delta: 2.0
      v_linear_min: 0.05
      v_linear_max: 0.2
      v_angular_max: 0.3
